### DISCUSSION TITLE
NymphCast Music Provider and Player Provider for "Music Assistant" (a.k.a. MA) for streaming high-quality audio directly to and from MA-streaming interoperability/integration software?

### DISCUSSION THREAD

**Discussion by Hedda** [music-assistant/support#2504] [Feature requests and ideas]
Has anyone considered looking into using NymphCast casting server and/or client player to and/or from Music Assistant?

https://github.com/MayaPosch/NymphCast

https://github.com/MayaPosch/NymphCast-MediaServer

Player/Client can be used on example ESP32 or Raspberry Pi:

https://github.com/MayaPosch/NymphCastAudio-ESP32

https://github.com/MayaPosch/NymphCast-Player

More here;

https://github.com/MayaPosch?tab=repositories&q=Nymph&type=&language=&sort=

# *What is NymphCast?*

*NymphCast is a software solution which turns your choice of hardware into an audio and video source for a television or (powered) speakers. It enables the streaming of audio and video over the network from a client device or media server, as well as the streaming of internet media to a NymphCast server, controlled by a client device, or directly on the receiver.*

*In addition, the server supports NymphCast apps (currently in preview status) written in AngelScript to extend the overall NymphCast functionality with e.g. 3rd party audio / video streaming protocol support on the server side, and customisable UIs in the client application.*

*NymphCast requires the use of a client device in all but the GUI ('Smart TV') mode. The server application (receiver) runs on a target device, while the client device functions as a remote control device:*

![image](https://github.com/music-assistant/hass-music-assistant/assets/6320001/d922237c-010a-478c-8a7c-446528fbde60)

*Client-side core functionality is provided through the NymphCast library.*

**Comment by OzGav** [comment]
I dont know anything about this but what does it do that the existing solutions don’t?

**Reply by Hedda** [reply]
I thought NymphCast client/server solutions could be interesting as alternative to Squeezelite for streaming to an ESP32 media player?

As for example asked here -> https://github.com/orgs/music-assistant/discussions/1338

Is Squeeze lite project still being developed for new and other purposes then its original intent or are new features just being hacked on? 

Based on their readme it sounds like a new alternative solution would not be a bad idea now:

https://github.com/sle118/squeezelite-esp32?tab=readme-ov-file#forewords

"**_More and more people seems to use this without a LMS server, just for BT, AirPlay or Spotify. It's fine but understand that squeezeliteESP32 is primarily a Logitech Media Server player and has been designed around that concept. All the others are add-ons stitched to it, so other modes have their shortcomings. So please make sure you read [this](https://github.com/sle118/squeezelite-esp32?tab=readme-ov-file#Additional-configuration-notes-from-the-Web-UI) before opening an issue_**"

So should we only continue to use Squeezelite if it that project is only in maintenance mode or maybe also look at alternative solutions?

https://en.wikipedia.org/wiki/Squeezelite

PS:  I also understand that AirPlay and Spotify protocols it uses are reverse-engineered so those can not either really be extended more.

**Reply by OzGav** [reply]
The answer would be that while squeezelite works we won't be putting any resources into alternatives at this time. If someone else wants to contribute a PR they are more than welcome.
